src/view/dataElements/xdmObject.jsx

```diff
@@ -10,403 +10,252 @@ OF ANY KIND, either express or implied. See the License for the specific languag
-            sandboxName: sandbox.name
-          })
-            .then(response => {
-              if (!response.results.length) {
-                throw new Error("No schemas found");
-              }
-              setSchemasMeta(response.results);
-              setSelectedSchemaMeta(undefined);
-              setSchemasMetaStatus(STATUS_LOADED);
-            })
-            .catch(() => {
-              setSchemasMetaStatus(STATUS_ERROR);
-            });
-        };
-
-        // TODO: address a suspected race condition where a previous selected item may return before
-        //  a secondary selected item
-        const onSchemaMetaSelected = schemaMeta => {
-          setSelectedSchemaMeta(schemaMeta);
-          if (schemaMeta) {
-            setSchemaOptionsSearch(schemaMeta.title);
-            setSchemaStatus(STATUS_LOADING);
-            fetchSchema({
-              orgId: options.initInfo.company.orgId,
-              imsAccess: options.initInfo.tokens.imsAccess,
-              schemaMeta,
-              sandboxName: selectedSandbox.name ? selectedSandbox.name : null
-            })
-              .then(schema => {
-                setSchemaStatus(STATUS_LOADED);
-                const initialValues = getInitialFormState({
-                  value: {},
-                  schema
-                });
-                options.resetForm(initialValues);
-              })
-              .catch(() => {
-                setSchemaStatus(STATUS_ERROR);
-              });
-          } else {
-            setSchemaStatus(STATUS_LOADED);
-          }
-        };
-
+      render={() => {
         return (
-          <XdmObject
-            {...options}
-            sandboxes={sandboxes}
-            selectedSandbox={selectedSandbox}
-            setSelectedSandbox={onSandboxSelected}
-            schemasMeta={schemasMeta}
-            schemasMetaStatus={schemasMetaStatus}
-            selectedSchemaMeta={selectedSchemaMeta}
-            setSelectedSchemaMeta={onSchemaMetaSelected}
-            schemaOptionsSearch={schemaOptionsSearch}
-            setSchemaOptionsSearch={setSchemaOptionsSearch}
-            schemaStatus={schemaStatus}
-            setSchemaStatus={setSchemaStatus}
+          <ExtensionViewForm
+            initialValues={{}}
+            getSettings={({ values }) => {
+              return childRef.current.getSettings({ values });
+            }}
+            validateFormikState={({ values }) => {
+              return childRef.current.validateFormikState({ values });
+            }}
+            validateNonFormikState={() => {
+              return childRef.current.validateNonFormikState();
+            }}
+            render={({ initInfo, resetForm }) => {
+              return (
+                <XdmObject
+                  ref={childRef}
+                  initInfo={initInfo}
+                  resetForm={resetForm}
+                />
+              );
+            }}
```

Comment:
I'd like to remove this `ref` stuff if there's a better way. Maybe it makes sense to not force the consumer of ExtensionViewForm to provide `initialValues`, `getSettings`, and validation methods up front, but instead provide a callback function through `render` or maybe use context to allow XDMObject to register its methods?

Reply:
It's kind of a chicken or the egg problem. If I render ExtensionViewForm inside the XDMObject, then the XDM Object won't have immediate access to `resetForm` which it uses in `useOnSandboxSelectionChange`, `useOnSchemaMetaSelectionChange`, and within a `useEffect`. I'll show you, and then show you the new API I've been playing with.